BetterTeams/src/main/java/com/booksaw/betterTeams/team/level/LevelManager.java

```diff
@@ -0,0 +1,106 @@
+package com.booksaw.betterTeams.team.level;
+
+import com.booksaw.betterTeams.Main;
+import org.bukkit.configuration.ConfigurationSection;
+import org.bukkit.configuration.file.FileConfiguration;
+
+import java.util.*;
+import java.util.logging.Level;
+
+public class LevelManager {
+
+	private static final Map<Integer, TeamLevel> levels = new HashMap<>();
+
+	/**
+	 * Reloads all levels from the config.yml file.
+	 */
+	public static void reload() {
+		levels.clear();
+		FileConfiguration config = Main.plugin.getConfig();
+		ConfigurationSection section = config.getConfigurationSection("levels");
+
+		if (section == null) {
+			Main.plugin.getLogger().log(Level.SEVERE, "Section 'levels' not found in config.yml! using default level 1.");
+			// fallback level to prevent crashes
+			levels.put(1, new TeamLevel(1, 10, 0, 0, 0, 1, 1, "0m", new ArrayList<>(), new ArrayList<>(), new ArrayList<>()));
+			return;
+		}
+
+		for (String key : section.getKeys(false)) {
+			try	{
+				int levelNum = Integer.parseInt(key.replace("l", ""));
+
+				int teamLimit = section.getInt(key + ".teamLimit", 10);
+				int maxChests = section.getInt(key + ".maxChests", 0);
+				int maxWarps = section.getInt(key + ".maxWarps", 0);
+				double maxBal = section.getDouble(key + ".maxBal", -1);
+				int maxAdmins = section.getInt(key + ".maxAdmins", -1);
+				int maxOwners = section.getInt(key + ".maxOwners", 1);
+
+				String price = section.getString(key + ".price", "0m");
+
+				List<String> startCmds = section.getStringList(key + ".startCommands");
+				List<String> endCmds = section.getStringList(key + ".endCommands");
+				List<String> lore = section.getStringList(key + ".rankLore");
+
+				TeamLevel teamLevel = new TeamLevel(
+						levelNum,
+						teamLimit,
+						maxChests,
+						maxWarps,
+						maxBal,
+						maxAdmins,
+						maxOwners,
+						price,
+						startCmds,
+						endCmds,
+						lore
+				);
+
+				levels.put(levelNum, teamLevel);
+
+			} catch (NumberFormatException e) {
+				Main.plugin.getLogger().warning("Invalid level key format in config (expected l1, l2...): " + key);
+			}
+		}
+	}
+
+
+	/**
+	 * Retrieves a specific level object.
+	 * @param level The level number.
+	 * @return The TeamLevel object
+	 */
+	public static TeamLevel getLevel(int level) {
+		return levels.getOrDefault(level, levels.get(1));
```

Comment:
Would this not make more sense to be nullable or throw an error rather than defaulting? Otherwise people may be getting the first level if they have some invalid logic (i.e. forgetting to check if the level exists)